server/models/study.py

```diff
@@ -176,6 +186,54 @@ def getImages(self, study):
             'meta.studyId': study['_id']}).distinct('meta.imageId')
         return Image().find({'_id': {'$in': imageIds}})
 
+    def addWaitingUser(self, study, user):
+        """
+        Add user to list of users waiting to participate in the study.
+        """
+        waitingUsers = study['meta'].get('waitingUsers', [])
+        waitingUsers.append(user['_id'])
+
+        # "setMetadata" will always save
+        self.setMetadata(
```

Comment:
I'd like to move away from `setMetadata`. Can you just do a `.save` or even a `.update` (with `$addToSet`)?